qiskit/backends/aer/qasm_simulator.py

```diff
@@ -146,13 +147,18 @@ def run(self, qobj):
         return aer_job
 
     def _run_job(self, job_id, qobj):
+        if isinstance(qobj, Qobj):
```

Comment:
Can you check if Is there currently (ie post `0.6`) a code path that leads to this function receiving something other than a `Qobj` (and one that does not have `qobj.config`)? If it's not the case, it might be cleaner to make the modifications directly on the object instead of performing a conversion back and forth to dict.

Reply:
I did a quick scan through the code and didn't see anything that passed it as a dict, but I wasn't 100% confident. So I figured it'd be safer to just keep everything assuming there could be a dict.

Reply:
Can you go further than a quick scan, for example running the tests with a version that assumes the input will always be a `Qobj`? If it passes, the small effort involved would in turn save efforts for future maintainers, as we would manage to forget about dicts in this function.

Reply:
Well that's part of the problem there are no tests that actually use the clifford simulator (which is why things pass today, because without this fix it clearly doesn't work). I only caught this because I added a test that tried to actually run on all the installed backends (see #1185 ). Personally I don't feel the maintenance burden for treating it as a dict is really high at all, but I'll try running #1185 with a modified version of this patch using just a QObj object.